blockmanager.go

```diff
@@ -2736,6 +2736,11 @@ func (b *blockManager) checkHeaderSanity(blockHeader *wire.BlockHeader,
 func (b *blockManager) calcNextRequiredDifficulty(newBlockTime time.Time,
 	reorgAttempt bool) (uint32, error) {
 
+	// bitcoind doesn't do retargeting on regtest, we do the same to ensure compatibility
```

Comment:
This looks a bit hacky indeed... `bitcoind` solved it by adding another boolean to the chain params: https://github.com/bitcoin/bitcoin/blob/e695d8536e534f1e59de4a62a0d87a93e7a73456/src/kernel/chainparams.cpp#L415
Perhaps we should do the same in `btcd`? Then we could solve it for both Neutrino and `btcd`.

Reply:
I have a PR up in `btcd` that can be re-used here once merged: https://github.com/btcsuite/btcd/pull/1985

Reply:
Missed that, Great!